### ISSUE TITLE
Dynamic topic publishing - Help

### ISSUE THREAD

**Issue by davydog187** [bbalser/elsa#78]
This is more of a request for help than an issue. We are publishing to topics that are decided at runtime, so configuring the producer ahead of time is not really feasible.

When we publish, we get these messages everytime:

```
15:44:07.515 [info]  client :elsa_default_client connected to localhost:9092
15:44:07.571 [info]  client :elsa_default_client connected to localhost:9092
15:44:07.621 [info]  client :elsa_default_client connected to localhost:9092
15:44:07.669 [info]  client :elsa_default_client connected to localhost:9092
```

My best understanding from looking at the producer code is that it is doing an "ad-hoc" produce: https://github.com/bbalser/elsa/blob/master/lib/elsa/producer.ex#L61

It's not clear from what's available in the documentation, if there is a way to dynamically register publishers. Guidance would be appreciated. Thank you!

**Comment by jeffgrunewald** [comment]
We don't have a mechanism built into Elsa to dynamically register producers under a single supervisor, but the way we've handled that in the past has been to have the runtime environment supply the necessary information to configure a producer and then stand up an elsa supervision tree (complete with producer) under a dynamic supervisor.

If you list of available topics is finite you could stand up one for each possible topic and then dispatch based on the name, but if that's entirely determined by the runtime environment then the first approach is what I would use.

How does that sound?

**Comment by davydog187** [comment]
Does that mean that every producer would have to maintain its own connection to Kafka? That sounds like a bit overkill

**Comment by jeffgrunewald** [comment]
IIRC we supported a shared connection in a much earlier version, but we ran into issues in one supervision tree crashing and taking down the shared connection which caused everything to come down. The overhead hasn't been an issue that we've really noticed but dedicating those connections has definitely made isolating failures less impactful.

**Comment by davydog187** [comment]
Gotcha. Ok that’s helpful, I’ll try it. Thanks!

**Comment by bbalser** [comment]
We have made significant changes to the internal architecture since that early version.  We should be able to support starting dynamic producers, I will look into adding it.

**Comment by jdenen** [comment]
This has been released as part of `v1.0.0-rc.1`.
